pywbemtools/pywbemcli/_cmd_statistics.py

```diff
@@ -0,0 +1,436 @@
+            to_on_off(desired_state)))
+        return
+
+    objmgr[OBJMGR_STAT_PROPERTY_NAME] = desired_state
+
+    # Otherwise modify the property in the server
+    try:
+        conn.ModifyInstance(objmgr, PropertyList=[OBJMGR_STAT_PROPERTY_NAME],
+                            IncludeQualifiers=False)
+    except Error as ce:
+        raise click.ClickException('Modification of GatherStatisticalData '
+                                   'property in instance {} on server failed: '
+                                   'Exception {}'.format(objmgr.path, ce))
+
+    modified_objmgr = conn.GetInstance(objmgr.path)
+
+    if modified_objmgr[OBJMGR_STAT_PROPERTY_NAME] != desired_state:
+        raise click.ClickException(
+            'The WBEM server reported success when changing the {} property '
+            'in its {} instance to {} but did not actually change it.'.
+            format(OBJMGR_STAT_PROPERTY_NAME, OBJMGR_CLN, interop_ns))
+
+    click.echo("Server GatherStatisticalData set {}".
+               format(to_on_off(desired_state)))
+
+
+###############################################################
+#         Statistics cmds action functions
+###############################################################
+
+
+def cmd_statistics_server_on(context):
+    """
+    Attempt to enable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, True)
+
+
+def cmd_statistics_server_off(context):
+    """
+    Attempt to disable the statistics gathering on the current server if there
+    is a current server defined.
+    """
+    test_conn_exists(context)
+    set_server_statistics(context, False)
+
+
+def cmd_statistics_status(context):
+    """
+    Show current status of both client and WBEM server statistics if there
+    is a connection.
+
+    It gets the status of the WBEM server statistics from the WBEM server
+    and the client statistics from the context and displays either a
+    table or text representation of the status.
+
+    If there is an issue with getting statistics information from the server,
+    it displays detailed information about the server response.
+
+    """
+
+    # validate output format with default of table
+    output_format = validate_output_format(context.output_format, ['TABLE',
+                                                                   'TEXT'])
+
+    test_conn_exists(context)
+
+    # Display server status as on/off
+    try:
+        objmgr = get_objmgr_inst(context)
+        try:
+            svr_status = objmgr[OBJMGR_STAT_PROPERTY_NAME]
+            svr_status = to_on_off(svr_status)
+        except KeyError:
+            svr_status = "No property {}".format(OBJMGR_STAT_PROPERTY_NAME)
+    except Error as er:
+        svr_status = "Not settable {}".format(er)
+
+    context.spinner_stop()
+
+    pywbemcli_stats_status = to_on_off(context.timestats)
```

Comment:
Determining the client status belongs before spinner_stop()

Reply:
Since that is a one line method that just changes True/false to on/off I was not worried and wanted to make it just before the result iused.

DONE No change